### DISCUSSION TITLE
Rails Console and Account table

### DISCUSSION THREAD

**Discussion by donnovan-lyons** [janko/rodauth-rails#80] [Q&A] [answered]
Hi, I'm new and relatively inexperienced with Ruby and Rails, although I've built out a few full-stack applications including a SPA with a jwt token,  and I'm currently trying to figure out how to implement rodauth-rails in my rails API backend. I have two questions:

First, I ran the basic config as outlined in your [tutorial](https://janko.io/adding-authentication-in-rails-with-rodauth/), i.e. `rails generate rodauth:install`. I got the following error after running `Account.new(email: "email@email.com")` in the rails console:

```
.rvm/gems/ruby-3.0.2/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require': cannot load such file -- rodauth/features/create_user (LoadError)
```
Any insight into what could be the problem here? 

Second question is that it seems that the the default configuration for the table that's created for a user is `Account` and all of the authentication features and configs as found in http://rodauth.jeremyevans.net/documentation.html has the same relationship with an `Account` table. The table I'm using for authentication is named `User`. Is there an easy way to rename everything so that it functions just fine with the current table i'm using?

Sorry if this is not the right forum to ask, but there's little documentation on how to use rodauth with an API.

**Comment by janko** [comment]
Hi Donnovan,

> Any insight into what could be the problem here?

Did you by any chance rename `:create_account` feature to `:create_user` in `enable` in your `RodauthApp`? If not, I would need to see your Rodauth config.

> The table I'm using for authentication is named `User`. Is there an easy way to rename everything so that it functions just fine with the current table i'm using?

Yes, Rodauth allows you change any table name via `*_table` configuration methods. The following configuration snippet should rename `account` with `user` in all table names:

```rb
methods.grep(/_table$/) do |table_method|
  public_send(table_method) { super().sub("account", "user") }
end
```

**Comment by donnovan-lyons** [comment, marked as answer]
Just want to add a small correction for future users here:

```
methods.grep(/_table$/) do |table_method|
     public_send(table_method) { super().to_s.sub("account", "user").to_sym }
end
```
